exasol/slc/internal/tasks/test/test_runner_db_test_with_export_task.py

```diff
@@ -0,0 +1,80 @@
+from typing import Dict, Generator
+
+from exasol_integration_test_docker_environment.lib.base.base_task import BaseTask
+from exasol_integration_test_docker_environment.lib.models.data.database_credentials import (
+    DatabaseCredentials,
+)
+from exasol_integration_test_docker_environment.lib.models.data.environment_info import (
+    EnvironmentInfo,
+)
+
+from exasol.slc.internal.tasks.export.export_containers import ExportFlavorContainer
+from exasol.slc.internal.tasks.test.test_runner_db_test_base_task import (
+    TestRunnerDBTestBaseTask,
+)
+from exasol.slc.internal.tasks.test.upload_exported_container import (
+    UploadExportedContainer,
+)
+from exasol.slc.models.export_info import ExportInfo
+
+
+class TestRunnerDBTestWithExportTask(TestRunnerDBTestBaseTask):
+
+    def register_required(self) -> None:
+        self.register_export_container()
+        self.register_spawn_test_environment()
+
+    def register_export_container(self) -> None:
+        export_container_task = self.create_child_task(
+            ExportFlavorContainer,
+            release_goals=[self.release_goal],
+            flavor_path=self.flavor_path,
+        )
+        self._export_infos_future = self.register_dependency(export_container_task)
+
+    def run_task(self) -> Generator[BaseTask, None, None]:
+        export_infos: Dict[str, ExportInfo] = self.get_values_from_future(
+            self._export_infos_future
+        )  # type: ignore
+        assert isinstance(export_infos, dict)
+        assert all(isinstance(x, str) for x in export_infos.keys())
+        assert all(isinstance(x, ExportInfo) for x in export_infos.values())
+
+        export_info = export_infos[self.release_goal]
```

Comment:
extract to _get_test_info()->TestInfo

class TestInfo
    target_name
    container_file
    is_new

Reply:
Done